NPE on mobile plugin in deps and no page.package set in config

cucumber.runtime.CucumberException: Failed to instantiate class ru.sbtqa.tag.stepdefs.en.MobileStepDefs
	at cucumber.runtime.java.DefaultJavaObjectFactory.cacheNewInstance(DefaultJavaObjectFactory.java:47)
	at cucumber.runtime.java.DefaultJavaObjectFactory.getInstance(DefaultJavaObjectFactory.java:33)
	at cucumber.runtime.java.JavaHookDefinition.execute(JavaHookDefinition.java:60)
	at cucumber.runtime.HookDefinitionMatch.runStep(HookDefinitionMatch.java:17)
	at cucumber.runner.UnskipableStep.executeStep_aroundBody0(UnskipableStep.java:22)
	at cucumber.runner.UnskipableStep$AjcClosure1.run(UnskipableStep.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at ru.sbtqa.tag.pagefactory.aspects.StashAspect.executeStep(StashAspect.java:67)
	at cucumber.runner.UnskipableStep.executeStep(UnskipableStep.java:22)
	at cucumber.api.TestStep.run_aroundBody0(TestStep.java:83)
	at cucumber.api.TestStep$AjcClosure1.run(TestStep.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at ru.sbtqa.tag.pagefactory.aspects.StashAspect.run(StashAspect.java:52)
	at cucumber.api.TestStep.run(TestStep.java:78)
	at cucumber.api.TestCase.run(TestCase.java:58)
	at cucumber.runner.Runner.runPickle(Runner.java:80)
	at cucumber.runtime.junit.PickleRunners$NoStepDescriptions.run(PickleRunners.java:140)
	at cucumber.runtime.junit.FeatureRunner.runChild(FeatureRunner.java:68)
	at cucumber.runtime.junit.FeatureRunner.runChild(FeatureRunner.java:23)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at cucumber.runtime.junit.FeatureRunner.run(FeatureRunner.java:73)
	at cucumber.api.junit.Cucumber.runChild(Cucumber.java:118)
	at cucumber.api.junit.Cucumber.runChild(Cucumber.java:56)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at cucumber.api.junit.Cucumber$1.evaluate(Cucumber.java:127)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
    Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at cucumber.runtime.java.DefaultJavaObjectFactory.cacheNewInstance(DefaultJavaObjectFactory.java:41)
	... 40 more
    Caused by: java.lang.NullPointerException
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:882)
	at com.google.common.reflect.ClassPath.getTopLevelClassesRecursive(ClassPath.java:162)
	at ru.sbtqa.tag.pagefactory.PageManager.getAllClasses(PageManager.java:153)
	at ru.sbtqa.tag.pagefactory.PageManager.cachePages(PageManager.java:132)
	at ru.sbtqa.tag.pagefactory.mobile.junit.MobileSetupSteps.initMobile(MobileSetupSteps.java:16)
	at ru.sbtqa.tag.pagefactory.mobile.junit.MobileStepsImpl.<init>(MobileStepsImpl.java:12)
	at ru.sbtqa.tag.pagefactory.mobile.junit.MobileSteps.<init>(MobileSteps.java:5)
	at java.lang.ThreadLocal$SuppliedThreadLocal.initialValue(ThreadLocal.java:284)
	at java.lang.ThreadLocal.setInitialValue(ThreadLocal.java:180)
	at java.lang.ThreadLocal.get(ThreadLocal.java:170)
	at ru.sbtqa.tag.pagefactory.mobile.junit.MobileSteps.getInstance(MobileSteps.java:10)
	at ru.sbtqa.tag.stepdefs.en.MobileStepDefs.<init>(MobileStepDefs.java:11)
	... 45 more

--- a/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/PageManager.java
+++ b/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/PageManager.java
@@ -1,6 +1,7 @@
 package ru.sbtqa.tag.pagefactory;
 
 import com.google.common.reflect.ClassPath;
+
 import java.io.IOException;
 import java.lang.reflect.Field;
 import java.lang.reflect.InvocationTargetException;
@@ -9,6 +10,7 @@ import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+
 import org.apache.commons.lang3.reflect.ConstructorUtils;
 import org.apache.commons.lang3.reflect.FieldUtils;
 import org.slf4j.Logger;
@@ -143,17 +145,20 @@ public class PageManager {
     }
 
     private static Set<Class<?>> getAllClasses() {
-        Set<Class<?>> allClasses = new HashSet();
+        Set<Class<?>> allClasses = new HashSet<>();
 
-        ClassLoader loader = Thread.currentThread().getContextClassLoader();
-        try {
-            for (ClassPath.ClassInfo info : ClassPath.from(loader).getTopLevelClassesRecursive(PROPERTIES.getPagesPackage())) {
-                allClasses.add(info.load());
+        if (PROPERTIES.getPagesPackage() != null) {
+            ClassLoader loader = Thread.currentThread().getContextClassLoader();
+            try {
+                for (ClassPath.ClassInfo info : ClassPath.from(loader).getTopLevelClassesRecursive(PROPERTIES.getPagesPackage())) {
+                    allClasses.add(info.load());
+                }
+            } catch (IOException ex) {
+                LOG.warn("Failed to shape class info set", ex);
             }
-        } catch (IOException ex) {
-            LOG.warn("Failed to shape class info set", ex);
+        } else {
+            LOG.warn("page.package property is not set");
         }
-
         return allClasses;
     }
 }

